request.go

```diff
@@ -356,8 +350,12 @@ func unmarshalNode(data *Node, model reflect.Value, included *map[string]*Node)
 				} else if v.Kind() == reflect.Int {
 					at = v.Int()
 				} else {
-					er = ErrInvalidTime
-					break
+					// Return error immediately to ensure a runtime panic doesn't swallow it.
+					return &ErrorObject{
```

Comment:
@shwoodard that is absolutely correct.

I was thinking that we could adopt this pattern throughout this pkg. It would allow consumers to immediately `MarshalErrors()` for any errors coming from this lib, and know that they will be well formatted and have sufficient information for communicating the error effectively and in a JSON API compatible manner. That is all future tense, of course.

Thoughts?